metabox/metabox/core/lxd_provider.py

```diff
@@ -308,8 +310,18 @@ def _run_transfer_commands(self, machine):
             self._transfer_file_preserve_mode(machine, src, dest)
 
     def _run_setup_commands(self, machine):
+        pre_cmds = []
+        if machine.config.revision != "current":
+            pre_cmds.append(
+                "bash -c 'cd /home/ubuntu/checkbox && "
+                "git clean -xfd && " # clean all untracked and changes to
```

Comment:
this is done on non-current revision, we have no way to know if the unstaged changes are actually compatible with the revision that this is creating, so it makes sense to clean them according to me. Additionally: this may make a test not fail due to an unstaged change making the revisions compatible while they are actually not

Reply:
> Additionally: this may make a test not fail due to an unstaged change making the revisions compatible while they are actually not

Because it is coded it this way.

What happens when you run metabox (not cross-version)? the code that's there get tested. I really dislike the idea of not using the code that's there as the thing that's tested against.

Reply:
Let's ask the question of what we're trying to achieve here.

We want to check whether the source code as it is right now will work with a previously landed revision.
Previously landed revision === main.
Current state === `find .`. Not `git show`.

So what we're aiming at is doing cross checks of:

main vs current state.
and
current state vs main.

The only git entity referenced in all of this is `main` which is constant and is not parametrised.